File: src/main/java/br/com/blz/testjava/http/data/response/WarehouseResponse.java
package br.com.blz.testjava.http.data.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class WarehouseResponse {
    private String locality; //Deveria ser um Enum?
    private Integer quantity;
    private WarehouseTypeResponse type;
}

File: src/main/kotlin/com/boticario/testkotlin/application/response/WarehouseResponse.kt
package com.boticario.testkotlin.application.response

import com.boticario.testkotlin.domain.entity.Warehouse
import com.boticario.testkotlin.domain.enum.WarehouseType

data class WarehouseResponse(
        val locality: String,
        val quantity: Int,
        val type: WarehouseType
) {
  companion object {
    fun fromDomain(warehouse: Warehouse) = WarehouseResponse(
      warehouse.locality,
      warehouse.quantity,
      warehouse.type
    )
  }
}
